sources/web/datalab/static.ts

```diff
@@ -112,6 +116,23 @@ function datalabFileExists(filePath: string) {
 }
 
 /**
+ * Sends a static 'custom.css' file located within the user's config directory.
+ *
+ * @param userId the ID of the current user.
+ * @param response the out-going response associated with the current HTTP request.
+ */
+function sendUserCustomTheme(userId: string, response: http.ServerResponse): void {
+    var customThemePath = path.join(settings.getUserConfigDir(userId), CUSTOM_THEME_FILE);
+    fs.stat(customThemePath, function(e, stats) {
```

Comment:
In this particular case, once we have the UI ready we can do a cache refreshing on an UI event (versus now, we cannot track when user modifies the config file). So we get both userbility and performance benefits.

For general handling of those types of files, it depends on how dynamic they are and if there is a way to track user actions so we know when to refresh. A disk read seems costly at per request level, especially on disks without SSD, so I would try to avoid that whenever possible.